### ISSUE TITLE
Particle not rendering correctly!

### ISSUE THREAD

**Issue by Atilgan0** [mob-sakai/ParticleEffectForUGUI#345]
**Describe the bug**
If a prefab containing ui particles is open before entering play mode, the particles are not rendered correctly when the game starts.

**To Reproduce**
1. Open prefab containing ui particle
2. Click play button
3. play any ui particle in the scene
4. See issue

**Screenshots**
![Screenshot_2](https://github.com/user-attachments/assets/93a5ed48-5ac9-47bf-882c-7d76a8dae6d5)

**Temp Solution**
Exit prefab mode on play mode state changed to ExitingEditMode. It was written editor code for this.

**Environment (please complete the following information):**
 - Version 4.10.3
 - Platform: Editor(Windows), Android
 - Unity version: 2022.3.53f1

**Comment by mob-sakai** [comment]
Thank you for your reporting!
Could you please attach a minimal project (included `Assets`, `Packages` and `ProjectSettings` directories) that reproduces the issue?
Of course, you don't need to include any secret assets from your project. Feel free to use free assets instead.

![repos](https://user-images.githubusercontent.com/12690315/130532459-0b5076f1-9b07-460b-b448-1d487e40ec1f.png)

**Comment by Atilgan0** [comment]
There is some news. 
I opened an empty project. I switched to Android platform. I tried to experience the problem here and noticed something like this. The problem occurs in the particle system that uses sprites in atlas. There is another particle in the scene that uses a normal sheet sprite. There is no problem with this. Also, we do not need to enter only the prefabs that have UIParticle in them, opening any prefab causes this problem. I zipped the project and added it below. I also added a video showing the problem.

Unity Version: 2022.3.53f1
Sprite Packer Mode: Sprite Atlas V2-Enabled
Platform: Android (windows pc)

I didin't change any other settings. All settings are in default.

[TestProject.zip](https://github.com/user-attachments/files/18150012/TestProject.zip)
